local-game.js

```diff
@@ -0,0 +1,512 @@
+	w_checker[i] = new checker(white_checker_class[i], "white", 2*i - 1 );
+	w_checker[i].setCoord(0,0);
+	block[2*i - 1].ocupied = true;
+	block[2*i - 1].pieceId = w_checker[i];
+}
+
+
+for (var i = 1; i <= 4; i++){
+	b_checker[i] = new checker(black_checker_class[i], "black", 56 + 2*i  );
+	b_checker[i].setCoord(0,0);
+	block[56 +  2*i ].ocupied = true;
+	block[56+  2*i ].pieceId =b_checker[i];
+}
+
+for (var i = 5; i <= 8; i++){
+	b_checker[i] = new checker(black_checker_class[i], "black", 40 +  2*i - 1 );
+	b_checker[i].setCoord(0,0);
+	block[ 40 + 2*i - 1].ocupied = true;
+	block[ 40 + 2*i - 1].pieceId = b_checker[i];
+}
+
+for (var i = 9; i <= 12; i++){
+	b_checker[i] = new checker(black_checker_class[i], "black", 24 + 2*i  );
+	b_checker[i].setCoord(0,0);
+	block[24 + 2*i ].ocupied = true;
+	block[24 + 2*i ].pieceId = b_checker[i];
+}
+
+
+the_checker = w_checker;
+
+function showMoves (piece) {
+	var match = false;
+	mustAttack = false;
+	if(selectedPiece){
+			erase_roads(selectedPiece);
+	}
+	selectedPiece = piece;
+	var i,j; 
+	for ( j = 1; j <= 12; j++){
+		if(the_checker[j].id == piece){
+			i = j;
+			selectedPieceindex = j;
+			match = true;
+		}
+	}
+
+	if(oneMove && !attackMoves(oneMove)){
+		changeTurns(oneMove);
+		oneMove = undefined;
+		return false;
+	}
+	if(oneMove && oneMove != the_checker[i] ){
+		return false;
+	}
+
+	if(!match) {
+	 return 0 ; 
+	}
+
+	
+	if(the_checker[i].color =="white"){
+		tableLimit = 8;
+		tableLimitRight = 1;
+		tableLimitLeft = 8;
+		moveUpRight = 7;
+		moveUpLeft = 9;
+		moveDownRight = - 9;
+		moveDownLeft = -7;
+	}
+	else{
+		tableLimit = 1;
+		tableLimitRight = 8;
+		tableLimitLeft = 1;
+		moveUpRight = -7;
+		moveUpLeft = -9;
+		moveDownRight = 9;
+		moveDownLeft = 7;
+	}
+ 	
+
+
+		attackMoves(the_checker[i]); 
+	
+
+	
+
+ 	if(!mustAttack){
+ 	  downLeft = checkMove( the_checker[i] , tableLimit , tableLimitRight , moveUpRight , downLeft);
+		downRight = checkMove( the_checker[i] , tableLimit , tableLimitLeft , moveUpLeft , downRight);
+		if(the_checker[i].king){
+			upLeft = checkMove( the_checker[i] , reverse_tableLimit , tableLimitRight , moveDownRight , upLeft);
+			upRight = checkMove( the_checker[i], reverse_tableLimit , tableLimitLeft , moveDownLeft, upRight)
+		}
+	}
+	if(downLeft || downRight || upLeft || upRight){
+			return true;
+		}
+	return false;
+	
+}
+
+
+function erase_roads(piece){
+	if(downRight) block[downRight].id.style.background = "#BA7A3A";
+	if(downLeft) block[downLeft].id.style.background = "#BA7A3A";
+	if(upRight) block[upRight].id.style.background = "#BA7A3A";
+	if(upLeft) block[upLeft].id.style.background = "#BA7A3A";
+}
+		
+
+
+function makeMove (index) {
+	var isMove = false;
+	if(!selectedPiece) 
+		return false;
+	if(index != upLeft && index != upRight && index != downLeft && index != downRight){
+		erase_roads(0);
+		selectedPiece = undefined;
+		return false;
+	}
+
+	if(the_checker[1].color=="white"){
+		cpy_downRight = upRight;

```

Comment:
I'm not sure I understand what's going on here :) Looks like a nice thing for optimisation, or, at least, adding comments